Source/VCGeneration/SplitAndVerifyWorker.cs

```diff
@@ -108,6 +108,10 @@ private void StartCheck(Split split, Checker checker)
           split.Stats, currentSplitNumber + 1, total, 100 * provenCost / (provenCost + remainingCost));
       }
 
+      if (options.XmlSink != null && DoSplitting) {
```

Comment:
Seems like something like an `interface SplitLogger { void StartSplit(..); void EndSplit(..)  }` with a ConsoleSplitLogger, XMLSplitLogger, and AggregateSplitLogger implementation could be useful to make the code a bit cleaner, but I can see that requires quite some refactoring which you may want to postpone.

Reply:
Totally agree, and yes not something I wanted to take on in this PR. Likely makes more sense to do that when fixing #460.